ga4gh/backend.py

```diff
@@ -222,18 +222,41 @@ def __init__(self):
         self._datasetIdMap = {}
         self._datasetIds = []
 
+    def _getDatasetFromReadsRequest(self, request):
+        if len(request.readGroupIds) != 1:
+            raise exceptions.NotImplementedException(
+                "Exactly one read group id must be specified")
+        compoundId = request.readGroupIds[0]
+        dataset = self._getDatasetFromCompoundId(compoundId)
+        return dataset
+
+    def _getDatasetFromVariantsRequest(self, request):
+        variantSetIds = request.variantSetIds
+        if len(variantSetIds) != 1:
+            raise exceptions.NotImplementedException(
+                "Exactly one variant set id must be specified")
+        compoundId = request.variantSetIds[0]
+        dataset = self._getDatasetFromCompoundId(compoundId)
+        return dataset
+
+    def _getDatasetFromCompoundId(self, compoundId):
+        splits = compoundId.split(':')
+        datasetId = splits[0]
+        dataset = self._datasetIdMap[datasetId]
```

Comment:
Shouldn't we check if the datasetId exists and raise a NotFoundException here? Also, we should add some test cases to test for this situation.